modules/apps/batch-engine/batch-engine-test/src/testIntegration/java/com/liferay/batch/engine/internal/exportimport/data/handler/test/BatchEnginePortletDataHandlerTest.java

```diff
@@ -687,11 +762,60 @@ private void _importLayouts(
 			exportImportConfiguration, file);
 	}
 
+	private void _inspectLARFile(
+			boolean attachmentIncludedInLARFile, long groupId, File larFile,
+			ObjectDefinition objectDefinition, ObjectEntry[] objectEntries,
+			String scope)
+		throws Exception {
+
+		try (ZipFile zipFile = new ZipFile(larFile)) {
+			ZipEntry zipEntry = zipFile.getEntry(
+				_getBatchFileNameWithPath(
+					objectDefinition.getName() + ".json", groupId));
+
+			if (zipEntry == null) {
+				throw new FileNotFoundException();
+			}
+
+			JSONArray jsonArray = JSONFactoryUtil.createJSONArray(
+				StringUtil.read(zipFile.getInputStream(zipEntry)));
+
+			Assert.assertEquals(objectEntries.length, jsonArray.length());
+
+			if (Objects.equals(ObjectDefinitionConstants.SCOPE_SITE, scope)) {
+				return;
+			}
+
+			for (ObjectEntry objectEntry : objectEntries) {
+				long dlFileEntryId = MapUtil.getLong(
+					objectEntry.getValues(), _OBJECT_FIELD_NAME_ATTACHMENT);
+
+				ZipEntry attachmentEntry = zipFile.getEntry(
+					"batch-binaries/" + dlFileEntryId);
+
+				if (attachmentIncludedInLARFile) {
+					Assert.assertNotNull(attachmentEntry);
+				}
+				else {
+					Assert.assertNull(attachmentEntry);
+				}
+			}
+		}
+	}
+
 	private void _testExportImportObjectEntriesToSameGroup(
-			Group group, String scope)
+			Group group, String scope, boolean attachmentVisible)
 		throws Exception {
 
-		ObjectDefinition objectDefinition = _addObjectDefinition(scope);
+		ObjectDefinition objectDefinition = null;
+
+		if (attachmentVisible) {
```

Comment:
What if we just add a new field to the object definition to have both visible and invisible attachments? This way we can simplify the code and have no conditions, just assertions.